clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,6 +259,7 @@ private void ensureNotFinished() {
     @Override
     public void close() throws IOException {
         if (!finished) {
+            flush();
```

Comment:
Do we really need to call `flush` here? It's a bit odd to have to do it twice. Would `writeBlock` be enough?